src/main/java/com/codeup/adlister/controllers/DeleteAdServlet.java

```diff
@@ -0,0 +1,42 @@
+package com.codeup.adlister.controllers;
+
+
+
+import com.codeup.adlister.dao.DaoFactory;
+import com.codeup.adlister.models.Ad;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+
+@WebServlet(name = "DeleteAdServlet", urlPatterns = "/ads/delete")
+public class DeleteAdServlet extends HttpServlet {
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+
+
+        long id = Long.parseLong(request.getParameter("id"));
+        Ad ad = DaoFactory.getAdsDao().getAd(id);
+        DaoFactory.getAdsDao().deleteAd(ad);
+
+        if (request.getSession().getAttribute("user") != null) {
```

Comment:
This conditional should be before line 22. Checking if the user is logged in should be the first action in this servlet. The redirect wouldn't be in the `else` section, but after the ad has been successfully deleted.